src/ioctl.rs

```diff
@@ -35,26 +56,37 @@ impl IoctlsBuilder {
     }
 }
 
-#[derive(Debug, Default, Eq, PartialEq)]
+/// A set of commands  commands that can be allowed with
+/// [`ioctl`](https://www.freebsd.org/cgi/man.cgi?query=ioctl) in capability
+/// mode.
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
 pub struct IoctlRights(Vec<u64>);
 
 impl IoctlRights {
     pub fn new(rights: Vec<u64>) -> IoctlRights {
         IoctlRights(rights)
     }
 
-    pub fn from_file<T: AsRawFd>(fd: &T, len: usize) -> CapResult<IoctlRights> {
+    /// Retrieve the list of currently allowed ioctl commands from a file.
+    ///
+    /// # Returns
+    ///
+    /// - `Ok(None)`:         All ioctl commands are allowed
```

Comment:
Instead of treating `Ok(None)` as sort of a "special" result, that seems a bit like a unintuitive use of `None`. Creating a better `IoctlRights` feels a bit more intuitive for the user.